### DISCUSSION TITLE
Seeed Studio XIAO ESP32S3 Sense

### DISCUSSION THREAD

**Discussion by JanickGers85** [arendst/Tasmota#20391] [Q&A] [answered]
Is the board Seeed Studio XIAO ESP32S3 Sense supported by Tasmota?
It seems a very cool hardware, would like to use it on some projects.

https://www.seeedstudio.com/XIAO-ESP32S3-Sense-p-5639.html

**Comment by sfromis** [comment, marked as answer]
Yes, just another small ESP32-S3 board, no problems. And there already exists a template for the board with camera add-on. https://templates.blakadder.com/seeedstudio_XIAO_ESP32S3_SENSE.html

As long as you config the pinout, there is basically little difference between boards with the same architecture. Sometimes it may be a question of whether PSRAM is configured, and IDK if this is the case here. Assuming that it was tested with camera, I'd also guess PSRAM to have been verified.

**Reply by JanickGers85** [reply]
Thanks a lot!
I've searched for the template but didn't find it, maybe it was late yesterday and I was too sleepy 😅😁

**Comment by barbudor** [comment]
Beware that the Seedstudio board requires an external antenna
You may also find interresting the TenstarRobit C3 Mini (on bord ceramic antenna) 
Slightly less cute but still small, the LuatOS C3 Core (on board Antena)

**Reply by JanickGers85** [reply]
Yep I know, it's a plus for the project I want to develop.
Thanks a lot for all the alternatives, I'll take a look for future project 👍

**Reply by sfromis** [reply]
While the ESP32-C3 is just fine for most Tasmota purposes, the ESP32-S3R8 in the Seeed board does provide 8MB of flash and also 8MB of PSRAM, which can be useful for some use cases.

**Reply by sfromis** [reply]
If you'd like onboard antenna, another very compact board with ESP32-S3 is from 01Space (also has a tiny screen). Lower flash/PSRAM than the Seeed board, but still a step up from the ESP32-C3.
https://github.com/01Space/ESP32-S3-0.42OLED

**Reply by JanickGers85** [reply]
Thanks a lot!
In your opinion which one performs the best with a RTSP stream?
I'm thinking about create a DIY security camera and with a Wrower I'm getting not so exciting results

**Reply by sfromis** [reply]
If what you want is a decent webcam, buy a webcam. While I'd recommend the S3 with PSRAM for Tasmota camera streaming, with those boards/sensors you still get low resolution or low framerate (and still not great image quality). Basically close to what you already get with WROVER, but maybe a bit better, as the S3 is a bit faster. Nothing major.

**Reply by barbudor** [reply]
I'd love to find a decent webcam that can be uncloudify
I have an old Tenvis camera, no PTZ, that was completely open. RSTP streamig, motion detection => upload frames to self configurable FTP, no need for mobile app, etc....
Now, any camera I bought recently on Ali needs an app for configuration (different app for each camera) and it's impossible to use outside of those apps.
If only there was a "Tasmota for IPcam..."

**Reply by sfromis** [reply]
"Alternative open firmware for your IP camera" looks relevant. Not tried it....
https://github.com/OpenIPC
The repositories there are updated very recently.

**Reply by JanickGers85** [reply]
Thanks @sfromis
The XIAO capability to handle a backup battery just blinded me and I've a few test with camera and ESP32 only after I placed my order 😅

@barbudor That's true, but there are some camera that are ONVIF/expose RTSP and you can use them in your LAN, like the Tapo C100/C110 (that I've just ordered).
You can register the webcam on the app the very first time, then once done you can block Internet access from your router.
I've done like that on one of my camera.

**Reply by btsimonh** [reply]
I'll hopefully be trying this out in a few days.  It has a mic at well as the camera - always wanted to try to  deliver the mjpeg and audio together in one stream, either in a custom mjpeg stream with javascript viewer or see if I can get rtsp to the browser somehow.  One blocker here is cors.
